x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalVerifier.java

```diff
@@ -20,20 +20,22 @@
 
 public final class LogicalVerifier extends PostOptimizationPhasePlanVerifier<LogicalPlan> {
 
-    public static final LogicalVerifier INSTANCE = new LogicalVerifier();
+    private LogicalVerifier(boolean isLocal) {
+        super(isLocal);
+    }
 
-    private LogicalVerifier() {}
+    public static LogicalVerifier getLocalVerifier() {
+        return new LogicalVerifier(true);
+    }
+
+    public static LogicalVerifier getGeneralVerifier() {
+        return new LogicalVerifier(false);
+    }
 
     @Override
-    boolean skipVerification(LogicalPlan optimizedPlan, boolean skipRemoteEnrichVerification) {
-        if (skipRemoteEnrichVerification) {
-            // AwaitsFix https://github.com/elastic/elasticsearch/issues/118531
-            var enriches = optimizedPlan.collectFirstChildren(Enrich.class::isInstance);
-            if (enriches.isEmpty() == false && ((Enrich) enriches.get(0)).mode() == Enrich.Mode.REMOTE) {
-                return true;
-            }
-        }
-        return false;
+    boolean hasRemoteEnrich(LogicalPlan optimizedPlan) {
+        var enriches = optimizedPlan.collectFirstChildren(Enrich.class::isInstance);
+        return enriches.isEmpty() == false && ((Enrich) enriches.get(0)).mode() == Enrich.Mode.REMOTE;
```

Comment:
Could you please help me understand why we only checking the first enrich?

Reply:
Tbh I am not sure. But I haven't changed that part, it's exactly the same as if were before. @bpintea may know more about this?

Reply:
https://www.elastic.co/docs/reference/query-languages/esql/esql-cross-clusters#esql-multi-enrich states: `"A _remote enrich command can’t be executed after a _coordinator enrich command. The following example would result in an error."`
This is checked `Enrich#checkForPlansForbiddenBeforeRemoteEnrich`

Reply:
This is confusing, but I think we only skip if the topmost ENRICH in the leftmost branch is remote - this is the only situation when the remote enrich hack can lead to shadowing issues that I can think of, because then the remote enrich is the top enrich node in the local plan.

We could skip more widely by looking for any remote enriches; but I think we want to fix the remote enrich hack soon, anyway, so maybe it's better to invest the effort in the real fix.